lib/dalli/ring.rb

```diff
@@ -46,12 +46,13 @@ def server_for_key(key)
       raise Dalli::RingError, "No server available"
     end
 
-    def lock
-      @servers.each(&:lock!)
+    def lock(servers)
```

Comment:
I don't know the underlying internals of lock but I assume that this dup call is will still lock what we want across threads calling it?

Reply:
yeah, the dup just makes a shallow copy because the caller mutates the list as servers return complete results. the underlying server objects are singletons